File: 백준/Silver/11899. 괄호 끼워넣기/괄호 끼워넣기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        char [] arr = br.readLine().toCharArray();
        int cnt = 0;
        int left = 0;

        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == ')') {
                if (left == 0) {
                    cnt++;
                } else {
                    left--;
                }
            } else {
                left++;
            }
        }
        cnt += left;
        System.out.println(cnt);
    }
}

File: 백준/Silver/11899. 괄호 끼워넣기/괄호 끼워넣기.py
def solution():
    S = input()

    cnt = 0
    left = 0

    for ch in S:
        if ch == ')':
            if left == 0:
                cnt += 1
            else:
                left -= 1
        else:
            left += 1
    cnt += left
    print(cnt)


solution()